src/main.ts

```diff
@@ -152,26 +127,27 @@ export default class VaultChat extends Plugin {
 			return []
 		}
 		const nearestVectors = this.vectorStore.getNearestVectors(embeddingResponse, 3, this.settings.relevanceThreshold)
-		const searchResults = Array.from(nearestVectors.keys())
-		const hydratedResults = []
-		for (const searchResult of searchResults) {
-			const abstractFile = this.app.vault.getAbstractFileByPath(searchResult)
-			if (!(abstractFile instanceof TFile)) {
-				console.warn(`Unexpected file type in search results. File: ${abstractFile?.name} `)
-				continue
+		const results = await Promise.all(nearestVectors.map(async (nearest, i) => {
+			let name = nearest.path.split('/').last() || ''
+			let contents = nearest.chunk
+			if (nearest.chunk && nearest.chunk.length) {
+				name = name + i // todo
+			}
+			if (!contents) {
+				const abstractFile = this.app.vault.getAbstractFileByPath(nearest.path) as TFile
+				const fileContentsOrEmpty = await this.app.vault.read(abstractFile)
+				let fileContents: string = fileContentsOrEmpty ? fileContentsOrEmpty : ''
+				if (fileContents.length > 1000) {
+					fileContents = `${fileContents.substring(0, 1000)}...`
```

Comment:
substring file to 1000 chats is a leftover from before. now that the request message also has chunks in it, this makes even less sense. We need to do the token based cut off instead.